src/Framework/Constraint/Count.php

```diff
@@ -65,6 +66,10 @@ protected function getCountOf($other)
                 return $this->getCountOfGenerator($iterator);
             }
 
+            if (!$iterator instanceof Iterator) {
+                return \iterator_count($iterator);
+            }
```

Comment:
Previously, this PR used a ternary when assigning `$key`. This guard conditional is likely clearer and avoids making an assumption about how a null value of `$key` will be handled later in the code (i.e. deciding whether to attempt rewinding the iterator).